devices/src/ioapic.rs

```diff
@@ -399,8 +399,16 @@ impl InterruptController for Ioapic {
     fn end_of_interrupt(&mut self, vec: u8) {
         for i in 0..NUM_IOAPIC_PINS {
             let entry = &mut self.reg_entries[i];
-            // Clear Remote IRR bit
             if vector(*entry) == vec && trigger_mode(*entry) == 1 {
+                // resamplefd write will be skipped due to split irqchip mode,
+                // so userspace EOI should handle the resamplefd write instead.
+                if let Some(rfd) = self
+                    .interrupt_source_group
+                    .unmask_notifier(i as InterruptIndex)
+                {
+                    rfd.write(1).unwrap();
+                }
```

Comment:
I wonder if there is a chance the same interrupt fired while processing this `eoi`?

Reply:
Found related comments and code from crosvm: https://github.com/google/crosvm/blob/main/devices/src/irqchip/ioapic.rs#L281-L286

Reply:
> I wonder if there is a chance the same interrupt fired while processing this `eoi`?

The hardware device's interrupt handler, vfio_intx_handler(), will mask the device's INTx interrupt, no interrupt from this device will be fired until the resamplefd is triggered. Once the resamplefd is triggered, the vfio_pci_intx_unmask_handler will be called to unmask the device's INTx interrupt again. So, before the "resamplefd" function is triggered, we will not see any interrupts. Please let me know if there are any errors in my understanding.